cmd/last-minute.go

```diff
@@ -76,34 +78,34 @@ type AccElem struct {
 	N     int64
 }
 
-// add dur to a as a single element.
+// Add a duration to a single element.
 func (a *AccElem) add(dur time.Duration) {
 	a.Total += int64(dur)
 	a.N++
 }
 
-// merge b into a.
+// Merge b into a.
 func (a *AccElem) merge(b AccElem) {
 	a.N += b.N
 	a.Total += b.Total
 }
 
-// avg converts total to average.
-func (a *AccElem) avg() uint64 {
+// Avg converts total to average.
+func (a AccElem) avg() uint64 {
 	if a.N >= 1 && a.Total > 0 {
 		return uint64(a.Total / a.N)
 	}
 	return 0
 }
 
-// LastMinuteLatencies keeps track of last minute latencies.
-type LastMinuteLatencies struct {
-	Totals  [60][sizeLastElemMarker]AccElem
+// lastMinuteLatency keeps track of last minute latency.
+type lastMinuteLatency struct {
+	Totals  [60]AccElem
```

Comment:
This is a breaking change. Since I assume this is used somewhere, I suspect a version has to be bumped somewhere.

Reply:
> This is a breaking change.

How ?  we still have LastMinuteLatencies, this is only internal change

Reply:
Oh for msgp, this is only used for internal communication. Umm, I think it is not a big deal to have weird values during rolling update of a cluster

Reply:
The serialized (msgp) representation is changed.

Before: `Totals [60][sizeLastElemMarker]AccElem`
After `Totals [60][]struct{[Totals: sizeLastElemMarker]lastMinuteLatency}`

I assume this is used somewhere, since it is included.

I guess this is would be compatible:

```
type LastMinuteLatencies [sizeLastElemMarker]lastMinuteLatency
``

Reply:
okay, let me check, thanks!

Reply:
> type LastMinuteLatencies [sizeLastElemMarker]lastMinuteLatency

I did this since it is simpler, but it is not msgp compatible. 

I don't think it is better to have better code than some transient errors during rolling update in this case

Reply:
@vadmeste If nothing else, it is more compact. since "Totals" key is omitted from each entry.

Reply:
> @vadmeste If nothing else, it is more compact. since "Totals" key is omitted from each entry.

Yes, this is done